pyvista/plotting/renderer.py

```diff
@@ -138,23 +144,12 @@ def camera_position(self, camera_location):
             return
         elif isinstance(camera_location, str):
             camera_location = camera_location.lower()
-            if camera_location == 'xy':
-                self.view_xy()
-            elif camera_location == 'xz':
-                self.view_xz()
-            elif camera_location == 'yz':
-                self.view_yz()
-            elif camera_location == 'yx':
-                self.view_yx()
-            elif camera_location == 'zx':
-                self.view_zx()
-            elif camera_location == 'zy':
-                self.view_zy()
-            else:
+            if camera_location not in self.CAMERA_STR_ATTR_MAP:
                 err = pyvista.core.errors.InvalidCameraError
                 raise err('Invalid view direction.  '
-                          'Use one of the following:\n'
-                          "    'xy', 'xz', 'yz', 'yx', 'zx', 'zy'")
+                          'Use one of the following:\n    %s'
+                          % ', '.join(self.CAMERA_STR_ATTR_MAP))
+            getattr(self, self.CAMERA_STR_ATTR_MAP[camera_location])()
```

Comment:
Typically, we should avoid `getattr` but its a pretty compelling use-case here.

One possible workaround is to move the `CAMERA_STR_ATTR_MAP` to the bottom of the class definition. That way all of the methods are defined. But having a class variable at the bottom of a class's definition is stylistically bad (IMO).

I'd be really hesitant to have methods referred to by a string name like this but since they are automatically tested, I think it is a pretty safe implementation.

Reply:
I really didn't like either implementation and was hoping there was another way around it.  I might research it more.

Reply:
I'll keep brainstorming this... 

We could refactor an introduce a new `Camera` class that holds all of the viewing methods and each renderer will own one. Then keep `CAMERA_STR_ATTR_MAP` on the renderer being aware of the methods it can call on the `Camera` and the `camera_position` setter for `Renderer` will be smarter than the setter for `Camera` allowing these strings.

But even that is all too elegant

Reply:
You could also use @staticmethod, @classmethod, but there isn't a way to access it without (), unless you write a decorator. You can also set it in __init__, check if it's None then set.